src/snowflake/connector/aio/_cursor.py

```diff
@@ -70,6 +70,11 @@ def __init__(
     def __aiter__(self):
         return self
 
+    def __iter__(self):
+        raise TypeError(
```

Comment:
should we use `NotImplementedError` here?

Reply:
same reason as above, this is to align the native python error experience if a class does not support sync iterator.
iterator operation contains network request (fetching result) so having a sync iterator doesn't make sense for async cursor